x/fungible/keeper/msg_server_udpate_zrc20_liquidity_cap.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"context"
+
+	cosmoserrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+	zetaObserverTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// UpdateZRC20LiquidityCap updates the liquidity cap for a ZRC20 token.
+func (k msgServer) UpdateZRC20LiquidityCap(goCtx context.Context, msg *types.MsgUpdateZRC20LiquidityCap) (*types.MsgUpdateZRC20LiquidityCapResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check authorization
+	if msg.Creator != k.observerKeeper.GetParams(ctx).GetAdminPolicyAccount(zetaObserverTypes.Policy_Type_group2) {
+		return nil, cosmoserrors.Wrap(sdkerrors.ErrUnauthorized, "update can only be executed by group 2 policy group")
+	}
+
+	// fetch the foreign coin
+	coin, found := k.GetForeignCoins(ctx, msg.Zrc20Address)
+	if !found {
+		return nil, types.ErrForeignCoinNotFound
+	}
+
+	// update the liquidity cap
+	coin.LiquidityCap = msg.LiquidityCap
```

Comment:
My understanding is that when `currentTotalSupply already > msg.LiquidityCap`, the ZRC20 will be capped immediately, correct?

Reply:
`currentSupply + amount > liquidityCap` is the condition checked, therefore setting a liquidity cap below current supply would immediately blocks the deposits